pandas/src/period.pyx

```diff
@@ -1203,6 +1208,7 @@ class Period(_Period):
 
 def _ordinal_from_fields(year, month, quarter, day,
                          hour, minute, second, freq):
+    freq = Period._maybe_convert_freq(freq)
```

Comment:
That's right. I was being overly cautious (maybe some other part of the code might need to call this in the future?) but I can remove it.